src/context.cpp

```diff
@@ -13,17 +13,79 @@
 #include <librealsense2/rs.h>              // RS2_API_FULL_VERSION_STR
 #include <src/librealsense-exception.h>
 
+#include <rsutils/os/special-folder.h>
+#include <rsutils/os/executable-name.h>
 #include <rsutils/easylogging/easyloggingpp.h>
 #include <rsutils/string/from.h>
 #include <rsutils/json.h>
 using json = nlohmann::json;
 
+#include <fstream>
+
+
+namespace librealsense {
+
+
+    static nlohmann::json load_config()
+    {
+        std::ifstream f( rsutils::os::get_special_folder( rsutils::os::special_folder::app_data ) + RS2_CONFIG_FILENAME );
+        if( ! f.good() )
+            return nlohmann::json::object();
+
+        // Load the realsense configuration file settings
+        try
+        {
+            return nlohmann::json::parse( f );
+        }
+        catch( std::exception const & e )
+        {
+            throw invalid_value_exception( "failed to load configuration file: " + std::string( e.what() ) );
+        }
+    }
+
+
+    static void merge_settings( nlohmann::json & settings, nlohmann::json const & overrides, std::string const & name )
+    {
+        if( ! overrides.is_object() )
+            throw invalid_value_exception( name + ": expecting an object; got " + overrides.dump() );
+        try
+        {
+            settings.merge_patch( overrides );
+        }
+        catch( std::exception const & e )
+        {
+            throw invalid_value_exception( "failed to merge " + name + ": " + std::string( e.what() ) );
+        }
+    }
+
+
+    static nlohmann::json load_settings( nlohmann::json const & context_settings )
+    {
+        // Allow ignoring of any other settings, global or not!
+        if( ! rsutils::json::get( context_settings, "inherit", true ) )
+            return context_settings;
+
+        auto config = load_config();
+
+        // Take the global 'context' settings out of the configuration
+        nlohmann::json settings;
+        if( auto global_context = rsutils::json::nested( config, "context" ) )
+            merge_settings( settings, global_context, "global config-file/context" );
```

Comment:
I thought that the parameter order should be switched.
As I understand specified `context_settings` should override executable settings that in turn override global setting

Reply:
That's exactly what it does
Here it take empty settings and "merges" the global, so we get global.